proxy/__init__.py

```diff
@@ -94,12 +91,10 @@ def get_blob(self, digest: str):
         resp = self.get(
             url,
             allow_redirects=True,
+            stream=True,
```

Comment:
Good question. The connection is kept open until the body is fully read, but it's unclear to me what happens if there's an error mid-way through reading the stream... I need to do some more digging.
See https://docs.python-requests.org/en/latest/user/advanced/#body-content-workflow

Reply:
It doesn't look like an error during read would cause the stream to close. I'm thinking we can add a finally to the try/except block calling `_download_blob`, calling either `stream.close()`[1], `stream.release_conn()`[2], or both (?).
What you think?
[1] https://urllib3.readthedocs.io/en/latest/reference/urllib3.response.html#urllib3.response.HTTPResponse.close
[2] https://urllib3.readthedocs.io/en/latest/reference/urllib3.response.html#urllib3.response.HTTPResponse.release_conn

Reply:
Yea, looks like you need to call close on errors. You could have a context manager to handle this for you.